src/Bicep.Core/Emit/EmitLimitationCalculator.cs

```diff
@@ -316,6 +317,25 @@ public static void DetectUnsupportedModuleParameterAssignments(SemanticModel sem
             }
         }
 
+        public static void DetectInvalidValueForParentProperty(SemanticModel semanticModel, IDiagnosticWriter diagnosticWriter)
+        {
+            foreach (var resourceDeclarationSymbol in semanticModel.Root.ResourceDeclarations)
+            {
+                if (resourceDeclarationSymbol.TryGetBodyPropertyValue(LanguageConstants.ResourceParentPropertyName) is { } referenceParentSyntax)
+                {
+                    if (referenceParentSyntax is ArrayAccessSyntax arrayAccess)
+                    {
+                        referenceParentSyntax = arrayAccess.BaseExpression;
+                    }
+
+                    if (semanticModel.ResourceMetadata.TryLookup(referenceParentSyntax) is not { })
+                    {
+                        diagnosticWriter.Write(DiagnosticBuilder.ForPosition(referenceParentSyntax.Span).InvalidValueForParentProperty());
```

Comment:
There's a slightly less verbose extension method you can use to do the same thing:
```
diagnosticWriter.Write(referenceParentSyntax, x => x.InvalidValueForParentProperty());
```